src/stores/RegionStore.ts

```diff
@@ -294,6 +313,17 @@ export class RegionStore {
         }
     };
 
+    @action focusCenter = (activeFrame: FrameStore) => {
+        if (activeFrame) {
+            activeFrame.setCenter(this.center.x, this.center.y);
+            
+            if (activeFrame.renderWidth < activeFrame.zoomLevel * this.boundingBox.x || activeFrame.renderHeight < activeFrame.zoomLevel * this.boundingBox.y) {
+                const zoomLevel = FOCUS_REGION_RATIO * Math.min(activeFrame.renderWidth / this.boundingBox.x, activeFrame.renderHeight / this.boundingBox.y);
+                activeFrame.setZoom(zoomLevel);
```

Comment:
Reset room level to let region width/height be about a maximum ration FOCUS_REGION_RATIO, which is 40% now.